@here/harp-datasource-protocol/lib/StyleSetEvaluator.ts

```diff
@@ -317,57 +369,72 @@ function removeDuplicatePropertyValues<T>(p: InterpolatedPropertyDefinition<T>)
     }
 }
 
-function createInterpolatedProperty<T>(prop: InterpolatedPropertyDefinition<T>) {
-    removeDuplicatePropertyValues(prop);
-    const propKeys = new Float32Array(prop.zoomLevels);
-    let propValues;
-    switch (typeof prop.values[0]) {
-        default:
-        case "number":
-            propValues = new Float32Array((prop.values as any[]) as number[]);
-            return {
-                interpolationMode:
-                    prop.interpolation !== undefined
-                        ? InterpolationMode[prop.interpolation]
-                        : InterpolationMode.Discrete,
-                zoomLevels: propKeys,
-                values: propValues,
-                exponent: prop.exponent
-            };
-        case "boolean":
-            propValues = new Float32Array(prop.values.length);
-            for (let i = 0; i < prop.values.length; ++i) {
-                propValues[i] = ((prop.values[i] as unknown) as boolean) ? 1 : 0;
+function procesStringEnocodedNumeralInterpolatedProperty(
+    format: StringEncodedNumeralFormat,
+    prop: InterpolatedPropertyDefinition<string>,
+    propValues: Float32Array,
+    maskValues: Float32Array
+): boolean {
+    let needsMask = false;
+
+    // Pixel and Meter values can be encoded together in the
+    // same array, so we need to evaluate the regExps for all
+    // the elements in the array.
+    if (
```

Comment:
Lot of duplication here. I think that adding more functions to `StringEncodedNumeralFormats` would reduce boilerplate here and there.